torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -259,12 +263,31 @@ void ProcessGroupAgent::sync() {
 }
 
 void ProcessGroupAgent::start() {
+  start_.store(true);
   listenerThread_ = std::thread(&ProcessGroupAgent::listenLoop, this);
 }
 
+void ProcessGroupAgent::localShutdown() {
+  if (!start_.load()) {
+    return;
+  }
+  LOG(INFO) << "Stopping ProcessGroupAgent.";
+  // CV to make sure we don't abort it in an invalid state.
+  {
+    std::unique_lock<std::mutex> lock(recvWorkMutex_);
+    recvWorkCV_.wait(lock, [this]() { return recvWork_; });
+    recvWork_->abort();
+    start_.store(false);
+  }
+  threadPool_.waitWorkComplete();
```

Comment:
Do we need to do this? Would prefer the local shutdown to be as fast as possible (but safe) and it shouldn't wait on outstanding operations.